openscm/adapters/__init__.py

```diff
@@ -29,72 +31,119 @@ class Adapter(metaclass=ABCMeta):
     _current_time: np.datetime64
     """Current time when using :func:`step`"""
 
-    _initialized: bool
-    """``True`` if model has been initialized via :func:`_initialize_model`"""
+    _parameters: ParameterSet
+    """Input parameter set"""
 
     _output: ParameterSet
     """Output parameter set"""
 
-    _parameters: ParameterSet
-    """Input parameter set"""
+    _parameter_views: Dict[HierarchicalName, ParameterInfo]
+    """Parameter views available with this model"""
+
+    _parameter_versions: Dict[HierarchicalName, int]
```

Comment:
No need for separeate versions as these are already stored in the views

Reply:
I think you still need them here as you need to know the version last time the adapter looked, not last time anyone looked through this view (which is also shared by the user who is setting things).